components/ws-manager/pkg/manager/manager.go

```diff
@@ -597,8 +597,7 @@ func (m *Manager) ControlPort(ctx context.Context, req *api.ControlPortRequest)
 		// outselves. Doing it ourselves lets us synchronize the status update with probing for actual availability, not just
 		// the service modification in Kubernetes.
 		wso := workspaceObjects{Pod: pod, PortsService: service}
-		err := m.completeWorkspaceObjects(&wso)
-		if err != nil {
+		if m.completeWorkspaceObjects(&wso) != nil {
```

Comment:
this change looks strange, next line will not return an error from call then? the same about next change below?

Reply:
Thanks for reviewing it and nice catch :slightly_smiling_face:! That is definitely not the behavior I wanted!

This problem happens 2 more changes I've made as ell